modules/user_accounts/php/my_preferences.class.inc

```diff
@@ -192,25 +192,23 @@ class My_Preferences extends \NDB_Form
             }
         }
         // END NOTIFICATIONS UPDATE
-
-        // make the set
         foreach ($values as $key => $value) {
-            if (!empty($value)) {
+            // Password updates are handled separately. Password_hash is removed
+            // from the initial update as otherwise it will be recorded in the
+            // history table unencrypted.
+            if (!empty($value) && $key != 'Password_hash') {
                 $set[$key] = $value;
             } else {
                 $set[$key] = null;
             }
         }
-
         // update the user
         $success = $user->update($set);
 
-        // prepend two random characters
-        if (isset($set['Password_hash'])) {
-            // Update CouchDB. Must do before password is salted/hashed.
-            $expiry = isset($values['Password_expiry'])
-                ? $values['Password_expiry'] : null;
-            $user->updatePassword($set['Password_hash'], $expiry);
+        // Now set the password. Note that this field is named incorrectly
```

Comment:
```suggestion
        // Now set the password. Note that this field is named incorrectly at this point in the code
```

to remove ambiguity (the database field stores a hash, it's just the UI form name that's not named correctly at this point)